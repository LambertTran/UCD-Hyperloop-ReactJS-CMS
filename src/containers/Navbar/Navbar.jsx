import React from 'react';

export default class Navbar extends React.Component{

  render() {
    return (
      <div className="c-navbar">
        <li>
          <i class="fa fa-tachometer" aria-hidden="true"></i>
          <a href="/admin/">DashBoard</a>
        </li>
        <li>
          <i class="fa fa-sign-out" aria-hidden="true"></i>
          <a href="/logout">Log out</a>
        </li>
      </div>
    )
  }
}