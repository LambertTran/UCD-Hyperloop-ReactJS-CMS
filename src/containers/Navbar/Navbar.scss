.c-navbar{
  display: flex;
  justify-content: space-between;
  background: black;
  li{
    list-style: none;
    margin: 0 20px 0 20px;
    padding: 10px;
    i{
      color: white;
      font-size: 1.6em;
      margin-right: 5px;
    }
    a{
      text-decoration: none;
      color: white;
      font-size: 1.2em;
    }
    @media screen and (min-width: 767px) {
      margin: 0 40px 0 40px;
    }
  }
}