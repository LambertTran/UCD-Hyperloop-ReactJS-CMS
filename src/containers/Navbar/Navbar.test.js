// packages
import React from 'react';
import { shallow } from 'enzyme';
import setupTests from '../../shares/setupTests';

// components
import Navbar from './Navbar';

describe('Navbar', () => {
  const navbar = shallow( <Navbar /> );

  it('renders correctly', () =>{
    expect(navbar).toMatchSnapshot();
  })
})