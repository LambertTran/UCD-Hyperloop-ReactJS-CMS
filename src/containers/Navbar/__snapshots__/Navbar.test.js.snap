// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <i
                    aria-hidden="true"
                    className="fa fa-tachometer"
          />
          <a
                    href="/admin/"
          >
                    DashBoard
          </a>
</li>,
        <li>
          <i
                    aria-hidden="true"
                    className="fa fa-sign-out"
          />
          <a
                    href="/logout"
          >
                    Log out
          </a>
</li>,
      ],
      "className": "c-navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              aria-hidden="true"
              className="fa fa-tachometer"
/>,
            <a
              href="/admin/"
>
              DashBoard
</a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "fa fa-tachometer",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DashBoard",
              "href": "/admin/",
            },
            "ref": null,
            "rendered": "DashBoard",
            "type": "a",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              aria-hidden="true"
              className="fa fa-sign-out"
/>,
            <a
              href="/logout"
>
              Log out
</a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "fa fa-sign-out",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log out",
              "href": "/logout",
            },
            "ref": null,
            "rendered": "Log out",
            "type": "a",
          },
        ],
        "type": "li",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <i
                        aria-hidden="true"
                        className="fa fa-tachometer"
            />
            <a
                        href="/admin/"
            >
                        DashBoard
            </a>
</li>,
          <li>
            <i
                        aria-hidden="true"
                        className="fa fa-sign-out"
            />
            <a
                        href="/logout"
            >
                        Log out
            </a>
</li>,
        ],
        "className": "c-navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                aria-hidden="true"
                className="fa fa-tachometer"
/>,
              <a
                href="/admin/"
>
                DashBoard
</a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "fa fa-tachometer",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DashBoard",
                "href": "/admin/",
              },
              "ref": null,
              "rendered": "DashBoard",
              "type": "a",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                aria-hidden="true"
                className="fa fa-sign-out"
/>,
              <a
                href="/logout"
>
                Log out
</a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "fa fa-sign-out",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log out",
                "href": "/logout",
              },
              "ref": null,
              "rendered": "Log out",
              "type": "a",
            },
          ],
          "type": "li",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
